pkg/agent/controller/networkpolicy/l7engine/reconciler.go

```diff
@@ -178,7 +178,7 @@ func generateTenantRulesData(policyName string, protoKeywords map[string]sets.Se
 	// Refer to Suricata detect engine in codebase for detailed tag keyword configuration.
 	var tagKeyword string
 	if enableLogging {
-		tagKeyword = " tag: session, 30, seconds;"
+		tagKeyword = " tag:host;"
```

Comment:
I wonder why the previous version was not working, it feels like `session` should also have included the request packet.

BTW, should we use something like `tag:host,1,packets` as we only need the first packet?

Reply:
I did some further investigation and found some problems:
Logging the wrong packet is a known bug/expected behavior documented https://redmine.openinfosecfoundation.org/issues/3480 and https://redmine.openinfosecfoundation.org/issues/2069. Basically the idea is that the log advanced past the first packet that triggered the event, so this will be the case for each `session`. If changed to `host`, the first alert stays the same, but the second alert would capture the initial packet as required.
Below is the complete log:
```
-----(trigger 1st time)
alert
10.10.1.3 → 10.10.1.4 TCP 80 → 47588 [RST, ACK] 
-----(trigger 2nd time)
10.10.1.4 → 10.10.1.3 TCP 55344 → 80 [SYN] SACK_PERM 
10.10.1.3 → 10.10.1.4 TCP 80 → 55344 [SYN, ACK] SACK_PERM 
10.10.1.4 → 10.10.1.3 TCP 55344 → 80 [ACK]
alert
10.10.1.4 → 10.10.1.3 HTTP GET /admin/index.html HTTP/1.1 
10.10.1.3 → 10.10.1.4 TCP 80 → 55344 [RST, ACK] 
alert: timeout
http: timeout
```
Given this fact, for the e2e it is possible to test it by triggering twice and look for the original packet. But I'm not sure of the necessity of `enableLogging` actually if this issue continues in Suricata.

Reply:
If the first packet is dropped and we have to wait for the next request, then I would argue that this is not a correct implementation of the feature. It's a different request in a different TCP session (source port is different).
So it looks like it is not possible to capture the correct packet at all?

Reply:
AFAIK if we use the existing Suricata log for this feature, it's not possible to capture that initial packet =(
I can remove the `tagged-packet`  if capturing the initial packet is the only purpose 😂

Reply:
Do you mean removing the feature completely?

Reply:
Actually I modified the `alert` configuration, it was able to log the initial packet
```
{"timestamp":"2024-09-06T22:56:25.262709+0000","flow_id":561336788716570,"in_iface":"antrea-l7-tap0","event_type":"alert","vlan":[2],"src_ip":"10.10.1.4","src_port":46094,"dest_ip":"10.10.1.3","dest_port":80,"proto":"TCP","pkt_src":"wire/pcap","tenant_id":2,"alert":{"action":"blocked","gid":1,"signature_id":1,"rev":0,"signature":"Reject by AntreaNetworkPolicy:default/allow-privileged-url-to-admin-role","category":"","severity":3,"tenant_id":2},"app_proto":"http","direction":"to_server","flow":{"pkts_toserver":3,"pkts_toclient":1,"bytes_toserver":307,"bytes_toclient":78,"start":"2024-09-06T22:56:25.261768+0000","src_ip":"10.10.1.4","dest_ip":"10.10.1.3","src_port":46094,"dest_port":80},"packet":"dtwWezuaHlOhfWpNgQAAAggARQAAjd+4QABABkSYCgoBBAoKAQO0DgBQH6jA1rH7p7qAGAH7zPsAAAEBCAou2X6GSIuizUdFVCAvYWRtaW4vaW5kZXguaHRtbCBIVFRQLzEuMQ0KSG9zdDogMTAuMTAuMS4zDQpVc2VyLUFnZW50OiBjdXJsLzcuNzQuMA0KQWNjZXB0OiAqLyoNCg0K","packet_info":{"linktype":1}}
```

But it applies to all rules because it's in the Suricata configuration, so cannot be controlled by `enableLogging`. Does it sound good to remove `enableLogging` which relates to `tagged-packet`, but add the `packet` support in `alert` that applies to all rules?

Reply:
Maybe it would be a good idea to discuss this at the community meeting.

Reply:
BTW, if this is for the `alert` event type, it only applies to rules that reject traffic, right?
We don't need the packet dump for allowed traffic, as we already have the app-layer metadata.

Reply:
Yes, tested that it only applies to rules rejecting traffic ( packet will only be added to `alert` logged for reject, only `http` will be logged for allow )

Reply:
We should do some perf testing using `ab` as @tnqn suggested, just to make sure that the number of rejected requests that we can handle stays roughly the same after enabling packet logging.